source/opt/inst_bindless_check_pass.cpp

```diff
@@ -563,10 +565,31 @@ void InstBindlessCheckPass::GenCheckCode(
     GenDebugStreamWrite(uid2offset_[ref->ref_inst->unique_id()], stage_idx,
                         {error_id, u_index_id, u_length_id}, &builder);
   }
+  // Generate a ConstantNull, converting to uint64 if the type cannot be a null.
+  if (new_ref_id != 0) {
+    analysis::TypeManager* type_mgr = context()->get_type_mgr();
+    analysis::Type* ref_type = type_mgr->GetType(ref_type_id);
+    if (ref_type->AsPointer() != nullptr) {
+      if (!get_feature_mgr()->HasCapability(spv::Capability::Int64)) {
+        std::unique_ptr<Instruction> cap_int64_inst(
+            new Instruction(context(), spv::Op::OpCapability, 0, 0,
+                            std::initializer_list<Operand>{
+                                {SPV_OPERAND_TYPE_CAPABILITY,
+                                 {uint32_t(spv::Capability::Int64)}}}));
+        get_def_use_mgr()->AnalyzeInstDefUse(&*cap_int64_inst);
+        context()->AddCapability(std::move(cap_int64_inst));
```

Comment:
I think you could replace these 7 lines with: context()->AddCapability(spv::Capability::Int64).

Reply:
IRContext::AddCapability() does the HasCapability() check internally so it doesn't need to be checked by the caller. I also fixed the other 2 places where the instrumentation code adds capabilities, one of which is for the same problem I'm fixing here.